### PR TITLE
Standardise container wiring for enclave and host

### PR DESCRIPTION
**Pull request by BedrockSquirrel**
### Why is this change needed?

- the way enclave and host processes are wired together and started is inconsistent at the moment
- wiring up and starting in the same big method makes it hard to stop and start a running server which I'm doing in some local testing for the resiliency stuff
- having a consistent approach between enclave and host means they can share logic for main methods, sim orchestration and agent/monitoring (if we go down that route in future)

### What changes were made as part of this PR:

- Create `Container` interface with a start/stop/status. This isn't its final form but gives us something to iterate on
- Create host_container and enclave_container to implement that interface and replace the "runner" packages
- Change main methods to use `Serve(container)` (main method usage unchanged so no script changes needed)
- Move wiring up of servers into their respective containers. (With intention to move more wiring there, more of a dependency-injection approach instead of the blackbox happening in Start methods at the moment)

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: integration/noderunner/noderunner_test.go [hunk 1]
```diff
@@ -83,8 +84,20 @@ func TestCanStartStandaloneObscuroHostAndEnclave(t *testing.T) {
...
 
-	go enclaverunner.RunEnclave(enclaveConfig)
-	go hostrunner.RunHost(hostConfig)
+	enclCont := enclavecontainer.NewEnclaveContainer(enclaveConfig)
+	go func() {
+		err := enclCont.Start()
+		if err != nil {
+			panic(err)
+		}
+	}()
```
Comment (otherview) [inline integration/noderunner/noderunner_test.go:93]: Doing this on-the-fly, but I think it works.

```suggestion
	go func() {
		err := enclavecontainer.NewEnclaveContainer(enclaveConfig).Start()
		if err != nil {
			panic(err)
		}
	}()
```

**Review by otherview — COMMENTED** [review]
Looks good. 
A couple of bits :
- Import ordering could be fixed https://github.com/uber-go/guide/blob/master/style.md#import-group-ordering
- Package names shouldn't repeat the parent package name host-> hostcontainer, typically host -> container is fine. But if it's confusing for some reason, I don't really mind keeping it as is.
- Container running should be derived from the health check I believe.

File: go/enclave/enclavecontainer/enclave_container.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	_, err := e.RPCServer.Stop(context.Background(), nil)
+	if err != nil {
+		e.Logger.Warn("unable to cleanly stop enclave", log.ErrKey, err)
+		return err
+	}
+	return nil
+}
+
+func (e *EnclaveContainer) Status() container.Status {
+	// todo: return recovery if DB is unavailable or if host is not keeping up with its responsibilities
```
[the remark this answers is not in the document]
Comment (BedrockSquirrel) [reply go/enclave/enclavecontainer/enclave_container.go:68]: As discussed, I'll get rid of the Status() for now until it's clear how we'll use it - pointlessly guessing the future. At least with the container we've got somewhere to put that sort of process-level checks and interactions

**Comment by BedrockSquirrel** [comment]
I think it might be mildly confusing to have three packages called "container" but they all have different method names so aliasing isn't too hard to keep on top of...

```
	enclavecontainer "github.com/obscuronet/go-obscuro/go/enclave/container"
	hostcontainer "github.com/obscuronet/go-obscuro/go/host/container"
```

**Pull request merged**